test/models/test_model_list_gp_regression.py

```diff
@@ -312,3 +314,31 @@ def test_fantasize(self):
             self.assertIsInstance(fm_i, SingleTaskGP)
             self.assertEqual(fm_i.train_inputs[0].shape, torch.Size([2, 8, 2]))
             self.assertEqual(fm_i.train_targets.shape, torch.Size([2, 8]))
+
+    def test_fantasize_with_outcome_transform(self):
+        with manual_seed(1234):
+            X = torch.rand(20, 1)
+
+        Y = -100 * X
+
+        # GP model
+        for outcome_transform in [Standardize(m=1), None]:
+            with self.subTest(outcome_transform=outcome_transform):
+                models_1d = [
+                    SingleTaskGP(
+                        X, Y[:, i].unsqueeze(-1), outcome_transform=outcome_transform
+                    )
+                    for i in range(Y.shape[-1])
+                ]
+                model = ModelListGP(*models_1d)
+                model.eval()
+
+                # Fantasy models
+                target_x = torch.tensor([[0.5]])
+                # TODO: make this more efficient
+                sampler = SobolQMCNormalSampler(1000, seed=9876)
+                fantasy_models = model.fantasize(target_x, sampler=sampler)
```

Comment:
Actually doing an MC estimate here is nice in the sense that it tests this code e2e, but as you say it's not very efficient.

One alternative to comparing the statistics here would be to just mock things. E.g. one idea would be to just mock the `get_fantasy_model` call [here](https://github.com/pytorch/botorch/blob/main/botorch/models/model_list_gp_regression.py#L98) and then check that the passed in `targets` are actually transformed versions of the ones passed into `condition_on_observations`. 

Or you could mock even more and just check that every model in the list that has an outcome transform ha the outcome transform called on the appropriate index of the passed in samples `Y`.

Reply:
Yeah it would be nice to have some more e2e MC checks, but I don't love it as a unit test due to the inefficiency and high potential to become flaky from minor numerical issues.